Sources/NIOPosix/SelectorWSAPoll.swift

```diff
@@ -0,0 +1,178 @@
+            }
+
+        let result = self.pollFDs.withUnsafeMutableBufferPointer { ptr in
+            WSAPoll(ptr.baseAddress!, UInt32(ptr.count), time)
+        }
+
+        if result > 0 {
+            // something has happened
+            for i in self.pollFDs.indices {
+                let pollFD = self.pollFDs[i]
+                guard pollFD.revents != 0 else {
+                    continue
+                }
+                // reset the revents
+                self.pollFDs[i].revents = 0
+                let fd = pollFD.fd
+
+                // If the registration is not in the Map anymore we deregistered it during the processing of whenReady(...). In this case just skip it.
+                guard let registration = registrations[Int(fd)] else {
+                    continue
+                }
+
+                var selectorEvent = SelectorEventSet(wsaEvent: pollFD.revents)
+                // in any case we only want what the user is currently registered for & what we got
+                selectorEvent = selectorEvent.intersection(registration.interested)
+
+                guard selectorEvent != ._none else {
+                    continue
+                }
+
+                try body((SelectorEvent(io: selectorEvent, registration: registration)))
+            }
+        } else if result == 0 {
+            // nothing has happened
+        } else if result == WinSDK.SOCKET_ERROR {
+            throw IOError(winsock: WSAGetLastError(), reason: "WSAPoll")
+        }
+    }
+
+    func register0(
+        selectableFD: NIOBSDSocket.Handle,
+        fileDescriptor: NIOBSDSocket.Handle,
+        interested: SelectorEventSet,
+        registrationID: SelectorRegistrationID
+    ) throws {
+        let poll = pollfd(fd: UInt64(fileDescriptor), events: interested.wsaPollEventSet, revents: 0)
+        self.pollFDs.append(poll)
+    }
+
+    func reregister0(
+        selectableFD: NIOBSDSocket.Handle,
+        fileDescriptor: NIOBSDSocket.Handle,
+        oldInterested: SelectorEventSet,
+        newInterested: SelectorEventSet,
+        registrationID: SelectorRegistrationID
+    ) throws {
+        fatalError("TODO: Unimplemented")
+    }
+
+    func deregister0(
+        selectableFD: NIOBSDSocket.Handle,
+        fileDescriptor: NIOBSDSocket.Handle,
+        oldInterested: SelectorEventSet,
+        registrationID: SelectorRegistrationID
+    ) throws {
+        fatalError("TODO: Unimplemented")
+    }
+
+    func wakeup0() throws {
+        // will be called from a different thread
+        let result = try self.myThread.withHandleUnderLock { handle in
+            QueueUserAPC(wakeupTarget, handle, 0)
+        }
+        if result == 0 {
+            let errorCode = GetLastError()
+            if let errorMsg = Windows.makeErrorMessageFromCode(errorCode) {
+                throw IOError(errnoCode: Int32(errorCode), reason: errorMsg)
+            }
+        }
+    }
+
+    func close0() throws {
+        // Throw away all pollFDs?
```

Comment:
Yeah, this is nice and straightforward to implement for `WSAPoll`, as there's no stateful kernel handle.